assets/js/app.js

```diff
@@ -49,4 +49,34 @@ var app = angular.module("Lucere", ["ngResource", "ngRoute", "dndLists"])
         templateUrl: "/js/templates/views/student_lesson.html"
       })
       .otherwise({redirectTo: "/"});
-  }]);
\ No newline at end of file
+  }]);
+  app.run(["$rootScope", "$location", "AuthService", "$route", function($rootScope, $location, AuthService, $route) {
+    $rootScope.$on("$locationChangeStart", function(event, next, current) {
+      AuthService.currentUser(function(user) {
+        // Check if the current user is trying to visit an admin route
+        var isAdminRoute = /(admin\/)/.test(next);
+        var paramArr = next.split("/");
+        var hasPermission;
+
+        if (isAdminRoute) {
+          var libId = paramArr[6];
+          hasPermission = user.administrating.some(function(adminLib) {
```

Comment:
The problem is that the user model does not have a direct reference to libraries.
Users have teams and teams have users.
A library has a team.
So what's going on here is that the route is giving the library id, and then the library id is being compared against the team id array found in the currentUser object.